gtime/hierarchical/bottom_up.py

```diff
@@ -0,0 +1,143 @@
+        >>> from gtime.time_series_models import AR
+        >>> from gtime.hierarchical import BottomUp
+        >>>
+        >>> testing.N, testing.K = 20, 1
+        >>> data1 = testing.makeTimeDataFrame(freq="s")
+        >>> data2 = testing.makeTimeDataFrame(freq="s")
+        >>> data = {'data1': data1, 'data2': data2}
+        >>> time_series_model = AR(p=2, horizon=3)
+        >>> tree = {'data1' : ['data2', 'data3'], 'data2': [], 'data3': []}
+        >>> hierarchical_model = BottomUp(model=time_series_model, hierarchy_tree=tree)
+        >>> hierarchical_model.fit(data)
+        >>> hierarchical_model.predict()
+    """
+    def __init__(self, model: BaseEstimator,
+                 hierarchy_tree: Union[str, Dict[str, Any]] = "infer"
+                 ):
+        super().__init__(model=model, hierarchy_tree=hierarchy_tree)
+
+    def fit(self, X: Dict[str, pd.DataFrame], y: pd.DataFrame = None):
+        """ Fit method
+
+        Parameters
+        ----------
+        X : Dict[str, pd.DataFrame], required
+            A dictionary of time series. Each is fitted independently
+        y : pd.DataFrame, optional, default = ``None``
+            only for compatibility
+
+        Returns
+        -------
+        self
+        """
+        self._check_is_dict_of_dataframes_with_str_key(X)
+        if self.hierarchy_tree_ == 'infer':
+            self._infer_hierarchy_tree(X)
+        self._initialize_models(X)
+        for key, time_series in X.items():
+            self.models_[key].fit(time_series)
+        return self
+
+    def predict(self, X: Dict[str, pd.DataFrame] = None):
+        """ Predict method
+
+        Parameters
+        ----------
+        X : Dict[str, pd.DataFrame], optional, default = ``None``
+            time series to predict. If ``None`` all the fitted time series are predicted.
+            The keys in ``X`` have to match the ones used to fit.
+
+        Returns
+        -------
+        predictions : Dict[str, pd.DataFrame]
+        """
+        check_is_fitted(self)
+        if X is None:
+            return self._predict_fitted_time_series()
+        else:
+            return self._predict_new_time_series(X)
+
+    def _initialize_models(self, X: Dict[str, pd.DataFrame]):
+        print(self.model)
+        self.models_ = {key: deepcopy(self.model) for key in X}
+
+    def _infer_hierarchy_tree(self, X: Dict[str, pd.DataFrame]):
+        self.hierarchy_tree_ = set(
+            X.keys()
+        )  # No need of a proper hierarchy tree for HierarchicalNaive
+
+    def _predict_fitted_time_series(self) -> Dict[str, pd.DataFrame]:
+        if self.hierarchy_tree_ =='infer':
+            return {key: model.predict() for key, model in self.models_.items()}
+        else:
+            return self._bottom_up_prediction_dictionary_creation(self)
+
+    def _bottom_up_prediction_dictionary_creation(self) -> Dict[str, pd.DataFrame]:
+        _bottom_up_dictionary = Dict[str, pd.DataFrame]
```

Comment:
Don't call it with underscore, since it is a variable limited to the function scope